payment-in-stablecoins/README.md

```diff
@@ -0,0 +1,252 @@
+
+### Detailed User Flow
+
+TokenBuyer is open to trading whenever:
+
+1. its USD balance is below the desired position, or
+2. the DAO has a debt towards a builder
+
+When TokenBuyer is open to trading, it offers trading bots the opportunity to sell their stablecoins (e.g. USDC) in exchange for ETH. Bots are incentivized by offering a price premium, providing them an arbitrage opportunity.
+
+Since acquiring USD from bots can take time, we expect the proposal page to clearly show proposal builders their outstanding balance and explain the balance should be settled shortly.
+
+TokenBuyer's USD position (baseline amount) configuration value helps mitigate debt delays: the higher the position, the less proposals will ever have to enter debt state.
+
+## Abstract
+
+This specification introduces two new contracts that facilitate stablecoin payments:
+
+1. `TokenBuyer`, a contract that:
+
+   1. allows anyone to sell it stablecoins in exchange for ETH with some additional incentive
+   2. funds the `Payer` contract with said stablecoins
+   3. auto-pays `Payer` debts upon receiving stablecoins
+
+2. `Payer`, a contract that:
+   1. allows the DAO to pay builders in stablecoins (e.g. USDC)
+   2. registers debt when the DAO's obligations exceed its stabecoin balance
+   3. allows anyone to run its outstanding debt payment function
+
+## Technical Specification
+
+### Overview
+
+#### Interaction flow
+
+1. A proposal is created, where the DAO wants to pay the builder $100K, with the following transactions
+   1. `Payer.sendOrRegisterDebt(recipient, 100_000)`
+   2. send TokenBuyer ETH at the amount of `TokenBuyer.ethNeeded(100_000, 2)` (assuming we want a 2x volatility buffer)
+2. Upon proposal execution:
+   1. Proposal recipient receives 100K stablecoins (e.g. USDC)
+   2. 133.33 ETH is sent from the treasury to `TokenBuyer` (assuming ETH/USD is $1500 and 2x buffer above)
+3. Arbitrageurs sell USD to `TokenBuyer` in exchange for ETH:
+   1. They call `TokenBuyer.buyETH(100_000)` (or smaller amounts across multiple txs)
+      1. Expected to happen when `TokenBuyer.price()` is better than other DEX prices
+   2. `Payer`'s USD balance grows by $100K, and an appropriate ETH balance is spent
+
+### `TokenBuyer` contract
+
+A new contract that helps the DAO acquire stablecoins by buying them from bots for a margin.
+
+#### Contract state variables
+
+Immutable:
+
+- `paymentToken` the ERC20 token to use as USD, e.g. USDC
+- `paymentTokenDecimalsDigits` as `10^paymentToken.decimals()`, used in conversions between token decimals and ETH decimals (18)
+
+Mutable:
+
+- `admin` contract admin, allowed to do certain lower risk operations
+- `payer` the `Payer` contract it helps fund with stablecoins
+- `priceFeed` the address of the `IPriceFeed` contract used to fetch stablecoin/ETH prices
```

Comment:
ETH/stablecoin instead of stablecoin/ETH

Reply:
thank you, fixed!